app/assets/javascripts/initializers/initializeBroadcast.js

```diff
@@ -83,17 +83,21 @@ function renderBroadcast(broadcastElement, data) {
 
 /**
  * A function to determine if a broadcast should render.
- * Does not render a broadcast on the `/new` route, if the current user
- * has opted-out, if the broadcast has already been inserted, or
- * if the key for the broadcast's title exists in localStorage.
+ * Does not render a broadcast on the `/new` route or the mod tools panel.
+ * Does not render a broadcast if the current user has opted-out,
+ * if the broadcast has already been inserted, or if the key for
+ * the broadcast's title exists in localStorage.
  *
  * If the broadcast exists in the DOM but was hidden by the articleForm,
  * the function will re-display it again by adding a class.
  *
  * @function initializeBroadcast
  */
 function initializeBroadcast() {
-  if (window.location.pathname === '/new') {
+  const pathname = window.location.pathname;
+  // The mod actions panel uses an iframe, which attempts to re-render the broadcast.
+  // By checking for `actions_panel` in the path, we can avoid this double render.
+  if (pathname === '/new' || pathname.includes('actions_panel')) {
```

Comment:
Hmmm... Instead of targeting actions_panel, is there a way to target an element? It seems like this issue happens on all `iframe`s we have and we do have several of them in the _sidecar_ in the Connect as well. Otherwise we will have to end up with 10s of URLs in that if statement. 

Connect issue: https://d.pr/v/boafUi

Reply:
![Cartman saying, "Oh crap, there's more of them!"](https://user-images.githubusercontent.com/11466782/86026448-0e85d780-b9f5-11ea-81c8-5a1f7209ec61.gif)

Reply:
I'm not totally sure if it is the right idea, but does it make more sense to render the broadcast on fewer pages (for example, the home page)? Rather than running the JS on every page, we could circumvent this issue by only using the broadcast on a small number of specific pages.

Reply:
@ludwiczakpawel The only way to really "target an element" here would be to check to see where the `<div id="active-broadcast" class="broadcast-wrapper"></div>` element is rendered. However, in both instances where the broadcast is being rendered—in the example of this bug, it is rendered once on the "article" page, and once on the "mod tools" page—the broadcast's `parentElement` is the `<body>` tag. The only thing that differentiates the two versions of the `<body>` tag is the `<div id="page-content"></div>` element, which is a child of the `<body>` tag.

You can see the difference between the two in the JS console. On the "article" page, the `data-current-page` attribute is `"stories-show"`, but on the "mod tools" page, that attribute is `"moderations-actions_panel"`. And that's about the only really "difference" in these two `<body>` tags.

The issue here now is that if we start targeting by an element (in this case, the only thing we could target by is the `data-current-page` on the `<div id="page-content"></div>` element), we run into the exact same issue that you were describing above. Every single place where we have an `iframe`, we now need to look at how that div's attribute changes, because that is the _only_ difference that really exists on the `<body>` tag. In other words, I understand your concern (and I am not a fan of this, either!) but targeting an element (the only element that we _can_ target) doesn't seem to really be much different in the result here.

@jacobherrington The announcement broadcasts are meant to be rendered site-wide (on every single page except for the `/new` route, where users write new articles). The issue with rendering the broadcast on fewer pages is that we'll then run into sort of the opposite problem as what @ludwiczakpawel described: we'll have to start figuring out the routes for where we _do_ want to render the broadcast, and then add them back in. So, it's sort of a question of do we want to be additive or subtractive here. If we take the additive approach, we're listing out all the places where the announcement _should _ render. If we're being subtractive, we're listing out all the places where the announcement _shouldn't_ render. 

IMHO, I think the real answer here is to move away from iframes since they:
1. have their own `window` context,
2. require more computational memory, and
3. can break browser's back/forward buttons + contribute to accessibility issues. 😬

Reply:
![image](https://user-images.githubusercontent.com/11466782/86030364-fd8b9500-b9f9-11ea-9110-abec84a15544.png)

Iframes are definitely hard to work around, but we've sort of adopted them... We will need to have a larger discussion if we choose to remove them long-term, but I agree we should have the conversation.

I personally prefer the additive approach you descrived because it's more explicit -- we are more likely to discover bugs because we know exactly where we are letting this code run. I was thinking of breaking this out of the initializePage file and running that JavaScript more intentionally in specific erb files, and I'm not clever enough to think of a good hack using selectors.

I totally get your point though, and I don't think it's the ideal solution...

If you feel good about it, in the short-term we can merge this, then re-visit a better solution?

Reply:
@jacobherrington HAHAHA that tweet! 😂  Thanks for the thoughtful reply. I do agree, we should consider having the conversation here as a larger topic. To be honest, part of the reason I thought that this fix was okay was that, while chatting with @Zhao-Andy, he mentioned that the mod tools will likely be re-written in Preact, which would eliminate this issue (but definitely won't solve it for other iframes 😬).

I'm open to the idea of breaking out the initialization logic into separate files and being more intentional with it if that seems easier (we'll just need to be cautious of making sure we cover all the many possible routes in our app where the broadcast would need to render). For the time being, I've added a `TODO` comment in the latest commit for @thepracticaldev/delightful just to make sure we come back to it (but if you end up doing this, feel free to scrap the comment).

TL;DR I appreciate your reply and agree that we want some kind of better solution here 😃